File: cpp/Medium/11_container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int width;
        int right = height.size()-1;
        int left = 0;
        int sum = 0;
        int preSum = 0;
        while(left < right){
            width = left - right;
            if( width < 0) width = width * -1;
            sum = width * min(height[left], height[right]);
            if( sum > preSum){
                preSum = sum;
            }
            if( height[left] < height[right]){
                left++;
            }
            else if( height[left] > height[right]){
                right--;
            }else{
                left++;
                right--;
            }
        }
        return preSum;
    }
};

File: python/medium/11_container_with_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        volume = 0
        while l < r:
            width = r - l
            min_height = min(height[l],height[r])
            if volume < min_height*width: volume = min_height*width
            if height[l] < height[r]: l+=1
            elif height[l] > height[r]: r-=1
            else:
                l+=1
                r-=1
        return volume

